.github/workflows/test.yml

```diff
@@ -162,3 +162,24 @@ jobs:
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
       - name: Image digest
         run: echo ${{ steps.docker_build.outputs.digest }}
+
+  autodeploy:
+    runs-on: ubuntu-latest
+    needs: [docker-deploy]
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/')
+    steps:
+    - uses: actions/checkout@v2
+    - name: Deploy Staging
+      if: github.ref == 'refs/heads/main'
+      run: bash scripts/autodeploy.sh
+      env:
+        AUTODEPLOY_URL: ${{ secrets.AUTODEPLOY_URL }}
+        AUTODEPLOY_TOKEN: ${{ secrets.AUTODEPLOY_TOKEN }}
+        TARGET_BRANCH: "staging"
+    - name: Deploy Develop
+      if: github.ref == 'refs/heads/develop'
+      run: bash scripts/autodeploy.sh
+      env:
+        AUTODEPLOY_URL: ${{ secrets.AUTODEPLOY_URL }}
+        AUTODEPLOY_TOKEN: ${{ secrets.AUTODEPLOY_TOKEN }}
+        TARGET_BRANCH: "develop"
```

Comment:
There is no `develop` branch.

Reply:
Yeah, it was in the case you are going to deploy develop in the future. At least the other projects that I checked they have it. I can remove it even so.

Reply:
About the background context, this is a change for the auto-deploy process, you had Dockerhub as the trigger but it didn't have auth method available, so because of security concerns we decide to use the Nginx auth of the auto-deploy server.

Reply:
> About the background context, this is a change for the auto-deploy process, you had Dockerhub as the trigger but it didn't have auth method available, so because of security concerns we decide to use the Nginx auth of the auto-deploy server.

The current Docker push action available: `docker/login-action@v1.13.0` has a combination of username and password. But an auth token can also be used instead of the password.

Wouldn't that be the same security-wise to the url+token combination?

Reply:
> Yeah, it was in the case you are going to deploy develop in the future. At least the other projects that I checked they have it. I can remove it even so.

I would remove it. Having a `develop` branch is a bigger change in our workflow so if we ever add it we'd also need to review the pipeline steps required.

Reply:
> I would remove it. Having a develop branch is a bigger change in our workflow so if we ever add it we'd also need to review the pipeline steps required.

I removed the develop branch

Reply:
> The current Docker push action available: docker/login-action@v1.13.0 has a combination of username and password. But an auth token can also be used instead of the password.

Wouldn't that be the same security-wise to the url+token combination?

Reply:
The main goal was to avoid the calls to the autodeploy server without authentication, now the requests (script file) has those credentials and make more secure this step